ports/stm32/flashbdev.c

```diff
@@ -109,6 +109,8 @@ STATIC byte flash_cache_mem[0x4000] __attribute__((aligned(4))); // 16k
 #define FLASH_SECTOR_SIZE_MAX (0x20000) // 128k max
 #define FLASH_MEM_SEG1_START_ADDR (0x08020000) // sector 1
 #define FLASH_MEM_SEG1_NUM_BLOCKS (256) // Sector 1: 128k / 512b = 256 blocks
+#define FLASH_MEM_SEG2_START_ADDR (0x08040000) // sector 2
+#define FLASH_MEM_SEG2_NUM_BLOCKS (256) // Sector 2: 128k / 512b = 256 blocks
```

Comment:
I think it's sufficient (and simpler) to just extend `FLASH_MEM_SEG1_NUM_BLOCKS` to 512.